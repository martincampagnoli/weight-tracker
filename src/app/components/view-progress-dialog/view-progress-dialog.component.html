<h2 mat-dialog-title>View Progress</h2>
<mat-dialog-content>
  <app-view-progress />
</mat-dialog-content>
