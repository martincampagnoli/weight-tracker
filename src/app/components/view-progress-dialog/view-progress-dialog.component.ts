import { ChangeDetectionStrategy, Component, inject } from '@angular/core';
import { MatButtonModule } from '@angular/material/button';
import { MatDialogTitle, MatDialogContent } from '@angular/material/dialog';
import { ViewProgressComponent } from '../view-progress/view-progress.component';

@Component({
  selector: 'view-progress-dialog',
  templateUrl: 'view-progress-dialog.component.html',
  standalone: true,
  imports: [
    MatButtonModule,
    MatDialogTitle,
    MatDialogContent,
    ViewProgressComponent,
  ],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class ViewProgressDialogComponent {}
