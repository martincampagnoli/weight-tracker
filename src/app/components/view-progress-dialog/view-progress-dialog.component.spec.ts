import { ComponentFixture, TestBed } from '@angular/core/testing';
import { MatDialogRef } from '@angular/material/dialog';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { provideMockStore } from '@ngrx/store/testing';

import { ViewProgressDialogComponent } from './view-progress-dialog.component';

describe('ViewProgressDialogComponent', () => {
  let component: ViewProgressDialogComponent;
  let fixture: ComponentFixture<ViewProgressDialogComponent>;
  let mockDialogRef: jest.Mocked<MatDialogRef<ViewProgressDialogComponent>>;

  const initialState = {
    data: [],
  };

  beforeEach(async () => {
    mockDialogRef = {
      close: jest.fn(),
      afterClosed: jest.fn(),
      beforeClosed: jest.fn(),
    } as any;

    await TestBed.configureTestingModule({
      imports: [ViewProgressDialogComponent, NoopAnimationsModule],
      providers: [
        provideMockStore({ initialState }),
        {
          provide: MatDialogRef,
          useValue: mockDialogRef,
        },
      ],
    }).compileComponents();

    fixture = TestBed.createComponent(ViewProgressDialogComponent);
    component = fixture.componentInstance;
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  describe('Component Creation', () => {
    it('should create', () => {
      expect(component).toBeTruthy();
    });

    it('should match snapshot', () => {
      fixture.detectChanges();
      expect(fixture).toMatchSnapshot();
    });
  });

  describe('Template Rendering', () => {
    it('should render ViewProgressComponent', () => {
      fixture.detectChanges();

      const viewProgressComponent =
        fixture.debugElement.nativeElement.querySelector('app-view-progress');
      expect(viewProgressComponent).toBeTruthy();
    });

    it('should render dialog title', () => {
      fixture.detectChanges();

      const dialogTitle =
        fixture.debugElement.nativeElement.querySelector('[mat-dialog-title]');
      expect(dialogTitle).toBeTruthy();
    });

    it('should render dialog content', () => {
      fixture.detectChanges();

      const dialogContent =
        fixture.debugElement.nativeElement.querySelector('mat-dialog-content');
      expect(dialogContent).toBeTruthy();
    });
  });
});
