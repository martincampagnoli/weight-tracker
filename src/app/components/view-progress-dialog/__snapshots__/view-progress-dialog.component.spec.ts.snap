// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewProgressDialogComponent Component Creation should match snapshot 1`] = `
<view-progress-dialog>
  <h2
    class="mat-mdc-dialog-title mdc-dialog__title"
    id="mat-mdc-dialog-title-a1"
    mat-dialog-title=""
  >
    View Progress
  </h2><mat-dialog-content
    class="mat-mdc-dialog-content mdc-dialog__content"
  >
    <app-view-progress>
      <p-chart
        class="p-chart"
        pc2=""
        style="display: block; position: relative;"
        styleclass="w-full"
        type="line"
      >
        <canvas
          role="img"
        />
      </p-chart>
    </app-view-progress>
  </mat-dialog-content>
</view-progress-dialog>
`;
